### ISSUE TITLE
[AI] When AI can't open a secure door, it stays there

### ISSUE THREAD

**Issue by Krzeszny** [Regalis11/Barotrauma#6459]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
This is a problem mostly with armories. AI will just stay there if it can't open a door.
[https://youtu.be/n56Gr7U1TvM](https://youtu.be/n56Gr7U1TvM)

**Steps To Reproduce**
Create a leak (tested) or break a machine (not tested) behind secured doors.

**Version**
v0.14.6.0

**I have 2 proposed fixes:**
a) easier – AI could ignore whatever is behind locked doors and find something else to do
b) better – AI should be able to open secure doors (configurable in settings) but always close them after getting through
When b) is disabled in settings by a player, a) should apply

**Comment by itchyOwl** [comment]
The bots shouldn't try to go where they can't access to. They should yell that they can't reach the place and continue with other things. Is that Humpback or some custom sub?

**Comment by itchyOwl** [comment]
Couldn't repro: ![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/ba13afa5-5457-4444-8610-6ddda8e8a6d0)

But there's clearly some issue in that video.

**Comment by itchyOwl** [comment]
Oh, and the access to the doors can already be configured in the editor: ![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/698af8ba-0922-4df7-95de-95676ebb2515)

**Comment by itchyOwl** [comment]
As a side note, there used to be a bug in the subeditor test mode where the crew couldn't access restricted doors they should have. I'm not sure if the issue still exists, but it could.

**Comment by Krzeszny** [comment]
> The bots shouldn't try to go where they can't access to. They should yell that they can't reach the place and continue with other things. Is that Humpback or some custom sub?

It's custom except for that inverted railgun I added. The damage was from a railgun shell falling back on the Humpback.

**Comment by itchyOwl** [comment]
The access is controlled by the sub designer and technically handled with id cards. I don't think we should introduce exceptions for the bots so that they could magically bypass the restrictions.

However, it's a bug if the bot gets stuck or attempts to access where they can't. But as I said, I couldn't reproduce that.

**Comment by UplinkPhobia** [comment]
I can confirm that I have the same problem on a custom sub. I checked the waypoints, they're pretty simple and similar to the ones on the screenshot above (and I tried moving them around a bit to see if it changed anything, but no). When there's a leak in armory, the bot assigned to leak fixing just runs into the door forever until I open it for them.
The door has a simple id_sec,id_cap security as well. 
![Capture](https://user-images.githubusercontent.com/24713223/140407428-0657cf17-abab-4754-91fc-332d2efbbc0f.png)
![Capture2](https://user-images.githubusercontent.com/24713223/140407451-4119fbf2-cab6-4a36-8057-7e1a1b4717bc.png)
![Capture3](https://user-images.githubusercontent.com/24713223/140408660-3019c2d8-0660-40c9-957e-b69ae6c52ad9.png)
The bots also seem to sometime "hesitate" and start going in the airlock below to try to go through the floor.
![Capture4](https://user-images.githubusercontent.com/24713223/140408596-7e207cc7-dec0-4fef-b2e8-3b3c43117447.png)
